File: shared/schemas/emergency.py
"""
Shared validation schemas for emergency cases
"""
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any


class LocationSchema(BaseModel):
    """Location schema"""
    lat: float = Field(..., ge=-90, le=90)
    lng: float = Field(..., ge=-180, le=180)
    address: Optional[str] = None


class EmergencyCaseSchema(BaseModel):
    """Emergency case schema"""
    case_id: str
    caller_id: Optional[str] = None
    emergency_type: str
    severity_level: str
    status: str
    location: Dict[str, Any]
    description: Optional[str] = None
    created_at: str
    updated_at: str

File: shared/types/emergency.ts
/**
 * Shared TypeScript types for emergency cases
 */

export enum EmergencyStatus {
  OPEN = "open",
  DISPATCHED = "dispatched",
  IN_PROGRESS = "in_progress",
  RESOLVED = "resolved",
  CANCELLED = "cancelled",
}

export enum SeverityLevel {
  CRITICAL = "critical",
  HIGH = "high",
  MEDIUM = "medium",
  LOW = "low",
}

export enum EmergencyType {
  ACCIDENT = "accident",
  MEDICAL = "medical",
  FIRE = "fire",
  CRIME = "crime",
  NATURAL_DISASTER = "natural_disaster",
  OTHER = "other",
}

export interface Location {
  lat: number;
  lng: number;
  address?: string;
}

export interface EmergencyCase {
  case_id: string;
  caller_id?: string;
  emergency_type: EmergencyType;
  severity_level: SeverityLevel;
  status: EmergencyStatus;
  location: Location;
  description?: string;
  created_at: string;
  updated_at: string;
}

